embedded-graphics/src/primitives/line/mod.rs

```diff
@@ -66,6 +67,68 @@ impl Dimensions for Line {
     }
 }
 
+/// Check signs of two signed numbers
+///
+/// Fastest ASM output compared to other methods. See: https://godbolt.org/z/zVx9cD
+fn same_signs(a: i32, b: i32) -> bool {
+    (a >= 0) == (b >= 0)
+}
+
+/// Intersection test result.
+#[derive(Copy, Clone, Debug)]
+pub enum Intersection {
+    /// Intersection at point
+    Point {
+        /// Intersection point.
+        point: Point,
+
+        /// The "outer" side of the intersection, i.e. the side that has the joint's reflex angle.
+        ///
+        /// For example:
+        ///
+        /// ```text
+        /// # Left outer side:
+        ///
+        ///  ⎯
+        /// ╱
+        ///
+        /// # Right outer side:
+        ///  │
+        /// ╱
+        /// ```
+        ///
+        /// This is used to find the outside edge of a corner.
+        outer_side: Side,
+    },
+
+    /// No intersection: lines are colinear or parallel.
+    Colinear,
+}
+
+/// Bresenham intersection result.
+#[derive(Debug, Copy, Clone)]
+pub(in crate::primitives) enum BresenhamIntersection {
+    /// Point
+    Point(Point),
+
+    /// Multiple intersection points.
+    ///
+    /// This may be caused by e.g. a shallow line where multiple points lie on the same Y
+    /// coordinate.
+    Line(Line),
+}
+
+impl BresenhamIntersection {
+    /// As a line
+    pub fn into_line(self) -> Line {
+        match self {
+            Self::Line(l) => l,
+            Self::Point(p) => Line::new(p, p),
+        }
+        .sorted_x()
+    }
+}
```

Comment:
Small optimization, because a single point line doesn't need to be sorted:
```suggestion
    pub fn into_line(self) -> Line {
        match self {
            Self::Line(l) => l.sorted_x(),
            Self::Point(p) => Line::new(p, p),
        }
    }
}
```